add error message in odata api log

zubair-arbi opened the pull request:
ENT-600
@saleem-latif @asadiqbal08 @douglashall

Add error message in `OData` API log.

zubair-arbi commented:
@mattdrayer Please have a look and let me know if this is what we wanted. I have explicitly included exception message in case we don't have `traceback` for the exception.

douglashall commented on common/djangoapps/third_party_auth/saml.py:
```diff
@@ -293,24 +293,28 @@ def get_user_details(self, attributes):
         try:
             client = self.get_odata_api_client(user_id=username)
             fields = ','.join(self.field_mappings)
+            odata_api_url = '{root_url}User(userId=\'{user_id}\')?$select={fields}'.format(
```
Were you planning to log the OData API URL below? If not, you can revert this change.

zubair-arbi replied:
I will revert this.

mattdrayer commented:
It would be useful to have the OData API URL available with the exception, because I can then test it using Postman outside of the system.

zubair-arbi commented:
@mattdrayer I have added the `OData` API URL in the log message.
@douglashall please have another look.
I have rebased and squashed from 0608b0bdb462e9b13c8bbd692decf29706cd3378
